testing/sliding-sync-integration-test/src/lib.rs

```diff
@@ -1035,4 +1035,182 @@ mod tests {
+
+        room.send(content, None).await?; // this should put our room up to the most recent
+
+        // let's subscribe
+
+        sync_proxy.subscribe(room_id.clone(), Default::default());
+
+        let mut room_updated = false;
+
+        for _n in 0..2 {
+            let room_summary = stream.next().await.context("sync has closed unexpectedly")?;
+            let summary = room_summary?;
+            // we only heard about the ones we had asked for
+            if summary.rooms.iter().any(|s| s == &room_id) {
+                room_updated = true;
+                break;
+            }
+        }
+
+        assert!(room_updated, "Room update has not been seen");
+
+        // force the pos to be invalid and thus this being reset internally
+        force_sliding_sync_pos(&sync_proxy, "100".to_owned());
+        let mut error_seen = false;
+        let mut room_updated = false;
+        for _n in 0..2 {
+            let summary = match stream.next().await {
+                Some(Ok(e)) => e,
+                Some(Err(e)) => {
+                    match e.client_api_error_kind() {
+                        Some(RumaError::UnknownPos) => {
+                            // we expect this to come through.
+                            error_seen = true;
+                            continue;
+                        }
+                        _ => Err(e)?,
+                    }
+                }
+                None => anyhow::bail!("Stream ended unexpectedly."),
+            };
+            // we only heard about the ones we had asked for
+            if summary.rooms.iter().any(|s| s == &room_id) {
+                room_updated = true;
+                break;
+            }
+        }
+
+        assert!(error_seen, "We have not seen the UnknownPos error");
+        assert!(room_updated, "Room update has not been seen");
+
+        // send another message
+
+        let room = client.get_joined_room(&room_id).context("No joined room {room_id}")?;
+
+        let content = RoomMessageEventContent::text_plain("Hello world");
+
+        let event_id = room.send(content, None).await?.event_id; // this should put our room up to the most recent
+
+        // let's see for it to come down the pipe
+        let mut room_updated = false;
+
+        for _n in 0..2 {
+            let room_summary = stream.next().await.context("sync has closed unexpectedly")?;
+            let summary = room_summary?;
+            // we only heard about the ones we had asked for
+            if summary.rooms.iter().any(|s| s == &room_id) {
+                room_updated = true;
+                break;
+            }
+        }
+        assert!(room_updated, "Room update has not been seen");
+
+        let Some(sliding_sync_room) = sync_proxy.get_room(room_id) else {
+            panic!("Sliding Sync room not found");
+        };
+
+        let Some(event) = sliding_sync_room.latest_event().await else {
+            panic!("No event found")
+        };
```

Comment:
These both look like good candidates for regular `.expect()`, or does that not work here?